custom/ibc-transfer/keeper/msg_server.go

```diff
@@ -0,0 +1,82 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	ibctransfermiddlewaretypes "github.com/notional-labs/composable/v6/x/ibctransfermiddleware/types"
+)
+
+type msgServer struct {
+	Keeper
+	bank      types.BankKeeper
+	msgServer types.MsgServer
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.MsgTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := k.Keeper.IbcTransfermiddleware.GetParams(ctx)
+	if params.ChannelFees != nil && len(params.ChannelFees) > 0 {
+		channelFee := findChannelParams(params.ChannelFees, msg.SourceChannel)
+		if channelFee != nil {
+			coin := findCoinByDenom(channelFee.AllowedTokens, msg.Token.Denom)
+			if coin != nil {
+				return &types.MsgTransferResponse{}, nil
+			}
+			minFee := coin.MinFee.Amount
+			charge := minFee
+			if charge.GT(msg.Token.Amount) {
+				charge = msg.Token.Amount
+			}
+
+			newAmount := msg.Token.Amount.Sub(charge)
+
+			if newAmount.IsPositive() {
```

Comment:
if it's not positive reject the transfer

Reply:
Custom ibc logic is going to charge all fee that possible. 
If the amount after fee is zero or less then we anywhere charge the fee and DO not proceed ibc transfer further but just ignore it.